cmd/operator/main.go

```diff
@@ -115,6 +149,9 @@ func main() {
 	flag.BoolVar(&enableLeaderElection, "leader-elect", false,
 		"Enable leader election for controller manager. "+
 			"Enabling this will ensure there is only one active controller manager.")
+	flag.BoolVar(&secureMetrics, "metrics-secure", false, "If set the metrics endpoint is served securely")
+	flag.BoolVar(&enableHTTP2, "enable-http2", false,
+		"If set, HTTP/2 will be enabled for the metrics and webhook servers")
```

Comment:
Here are 3 different ways to phrase the same thing: `Enable `, `If set `, `If set, `.  Maybe use:
```suggestion
		"Enable leader election for controller manager. "+
			"Enabling this will ensure there is only one active controller manager.")
	flag.BoolVar(&secureMetrics, "metrics-secure", false, "Enable API server authentication/authorization for the metrics endpoint")
	flag.BoolVar(&enableHTTP2, "enable-http2", false,
		"Enable HTTP/2 for the metrics and webhook servers (instead of using HTTP/1.1)")
```

Reply:
These were copied directly from the kubebuilder/operator-sdk's proposed code change.

I agree with your proposals with some modifications:
> Enable API server authentication/authorization for the metrics endpoint

I would say "Enable role based authentication.." because while the API server is used to query auth information, it doesn't decide whether someone should be granted access or not.

For:
>Enable HTTP/2 for the metrics and webhook servers (instead of using HTTP/1.1)

I would drop the "instead of using HTTP" part. Anyone configuring a kubernetes operator should in my mind have general knowledge of HTTP protocol and its versions. At least to the extent that there are multiple versions of the protocol.
And I'm also inclined to say "webhook services", as it's one "server" with two endpoints/services.

Reply:
`If set` comes from `kubebuilder` when scaffolding a project but I tend to agree @eero-t's suggestions make sense.

Reply:
> I would drop the "instead of using HTTP" part. Anyone configuring a kubernetes operator should in my mind have general knowledge of HTTP protocol and its versions. At least to the extent that there are multiple versions of the protocol.

Do you know the HTTP versions: https://en.wikipedia.org/wiki/HTTP#Summary_of_HTTP_milestone_versions ? :-)

My point was that the option is for _upgrading_ v1.1 to v2, instead of _downgrading_ v3 to v2...

Reply:
It's not upgrading. It's allowing HTTP/2 use. A client that doesn't understand HTTP/2 would still work with 1.1.